rotkehlchen/globaldb/handler.py

```diff
@@ -2008,3 +2010,93 @@ def get_exchange_name_from_assetid(exchange: Location, asset_identifier: str) ->
             ).fetchone()
 
         return None if identifier is None else identifier[0]
+
+    @staticmethod
+    def query_location_asset_mappings(
+            filter_query: 'LocationAssetMappingsFilterQuery',
+    ) -> tuple[list[dict[str, str | Location | None]], int, int]:
+        """Returns a tuple with the mappings, their amount according to the filter_query, and total
+        amount without any filter. Mappings are in the form of a list of dicts with the keys,
+        asset, location, and location_symbol."""
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            mappings_total = cursor.execute(
+                'SELECT COUNT(*) FROM location_asset_mappings',
+            ).fetchone()[0]
+
+            query, bindings = filter_query.prepare(with_pagination=False)
+            mappings_count = cursor.execute(
+                f'SELECT COUNT(*) FROM location_asset_mappings {query}', bindings,
+            ).fetchone()[0]
+
+            query, bindings = filter_query.prepare()
+            cursor.execute(
+                f'SELECT local_id, location, exchange_symbol FROM location_asset_mappings {query}', bindings,  # noqa: E501
+            )
+            return [{
+                'asset': identifier,
+                'location': None if location is None else str(Location.deserialize_from_db(location)),  # noqa: E501
+                'location_symbol': symbol,
+            } for identifier, location, symbol in cursor], mappings_count, mappings_total
+
+    @staticmethod
+    def add_location_asset_mappings(entries: list[LocationAssetMappingUpdateEntry]) -> None:
+        """Adds the given mapping entries of asset identifiers and their symbols in the given
+        location to the location_asset_mappings table.
+
+        May Raise: InputError if the pair of location and exchange_symbol already exist"""
+        with GlobalDBHandler().conn.write_ctx() as cursor:
+            for entry in entries:
+                try:
+                    cursor.execute(
+                        'INSERT INTO location_asset_mappings(local_id, location, exchange_symbol) VALUES(?, ?, ?)', (  # noqa: E501
+                            entry.asset.serialize(),
+                            None if entry.location is None else entry.location.serialize_for_db(),
+                            entry.location_symbol,
+                        ),
+                    )
+                except sqlite3.IntegrityError as e:
+                    raise InputError(
+                        f'Failed to add the location asset mapping of {entry.location_symbol} in '
+                        f'{entry.location} because it already exists in the DB.',
+                    ) from e
+
+    @staticmethod
+    def update_location_asset_mappings(entries: list[LocationAssetMappingUpdateEntry]) -> None:
+        """Updates the mapped asset identifiers in the location_asset_mappings table against their
```

Comment:
I already asked you to fix this last review, you marked it as done and it's not. Please don't do that.

```suggestion
        """Updates the mapped asset identifiers in the location_asset_mappings table based on their
```